src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -4055,28 +3990,15 @@ int32_t NaClSysCondTimedWaitAbs(struct NaClAppThread     *natp,
     goto cleanup;
   }
 
-  /* TODO(gregoryd): validate ts - do we have a limit for time to wait? */
-
-  cv_desc = NaClGetDesc(nap, cond_handle);
-  if (!cv_desc) {
-    retval = -NACL_ABI_EBADF;
-    goto cleanup;
-  }
-
-  mutex_desc = NaClGetDesc(nap, mutex_handle);
-  if (!mutex_desc) {
-    NaClDescUnref(cv_desc);
-    retval = -NACL_ABI_EBADF;
-    goto cleanup;
-  }
+  retval = lind_cond_timedwait(cond_handle, mutex_handle, (struct timespec*) &trusted_ts, nap->cage_id);
 
-  retval = ((struct NaClDescVtbl const *)cv_desc->base.vtbl)->TimedWaitAbs(cv_desc, mutex_desc, &trusted_ts);
-  NaClDescUnref(cv_desc);
-  NaClDescUnref(mutex_desc);
 cleanup:
   return retval;
 }
 
+//TODO: semaphores currently do not have a proper implementation as the desc numbers cannot be
```

Comment:
You mentioned that these aren't reachable because they're not implemented in our glibc, is that correct? If so add that here.